homeassistant/components/hue/__init__.py

```diff
@@ -82,135 +78,133 @@ def setup(hass, config):
+            return
+
+        await async_setup_bridge(
+            hass, discovery_info['host'],
+            'phue-{}.conf'.format(discovery_info['serial']))
+
+    discovery.async_listen(hass, SERVICE_HUE, async_bridge_discovered)
 
     # User has configured bridges
     if CONF_BRIDGES in conf:
         bridges = conf[CONF_BRIDGES]
+
     # Component is part of config but no bridges specified, discover.
     elif DOMAIN in config:
         # discover from nupnp
-        hosts = requests.get(API_NUPNP).json()
-        bridges = [{
+        websession = aiohttp_client.async_get_clientsession(hass)
+
+        async with websession.get(API_NUPNP) as req:
+            hosts = await req.json()
+
+        # Run through config schema to populate defaults
+        bridges = [BRIDGE_CONFIG_SCHEMA({
             CONF_HOST: entry['internalipaddress'],
             CONF_FILENAME: '.hue_{}.conf'.format(entry['id']),
-        } for entry in hosts]
+        }) for entry in hosts]
+
     else:
         # Component not specified in config, we're loaded via discovery
         bridges = []
 
-    for bridge in bridges:
-        filename = bridge.get(CONF_FILENAME)
-        allow_unreachable = bridge.get(CONF_ALLOW_UNREACHABLE)
-        allow_in_emulated_hue = bridge.get(CONF_ALLOW_IN_EMULATED_HUE)
-        allow_hue_groups = bridge.get(CONF_ALLOW_HUE_GROUPS)
-
-        host = bridge.get(CONF_HOST)
-
-        if host is None:
-            host = _find_host_from_config(hass, filename)
+    if not bridges:
+        return True
 
-        if host is None:
-            _LOGGER.error("No host found in configuration")
-            return False
-
-        setup_bridge(host, hass, filename, allow_unreachable,
-                     allow_in_emulated_hue, allow_hue_groups)
+    await asyncio.wait([
+        async_setup_bridge(
+            hass, bridge[CONF_HOST], bridge[CONF_FILENAME],
+            bridge[CONF_ALLOW_UNREACHABLE], bridge[CONF_ALLOW_HUE_GROUPS]
+        ) for bridge in bridges
+    ])
 
     return True
 
 
-def bridge_discovered(hass, service, discovery_info):
-    """Dispatcher for Hue discovery events."""
-    if "HASS Bridge" in discovery_info.get('name', ''):
-        return
-
-    host = discovery_info.get('host')
-    serial = discovery_info.get('serial')
-
-    filename = 'phue-{}.conf'.format(serial)
-    setup_bridge(host, hass, filename)
-
-
-def setup_bridge(host, hass, filename=None, allow_unreachable=False,
-                 allow_in_emulated_hue=True, allow_hue_groups=True,
-                 username=None):
+async def async_setup_bridge(
+        hass, host, filename=None,
+        allow_unreachable=DEFAULT_ALLOW_UNREACHABLE,
+        allow_hue_groups=DEFAULT_ALLOW_HUE_GROUPS,
+        username=None):
     """Set up a given Hue bridge."""
+    assert filename or username, 'Need to pass at least a username or filename'
+    # Resolve host to IP
+    resolved_host = await hass.async_add_job(socket.gethostbyname, host)
```

Comment:
Actually, thinking now, I don't think that we ever get a non-IP back from discovery. I think that we can remove this check and just use `host` as the identifier. We will require people to enter an ip address for host in the config.

Reply:
The reason the frontend doesn't start up is because it waits for the HOMEASSISTANT_START event but we don't fire that event until all setups are done.

https://github.com/home-assistant/home-assistant/blob/d5612b5ccc4b30ed62022a5b042459ac03425b1e/homeassistant/components/http/__init__.py#L146